### PR TITLE
Add new workflow

### PR DESCRIPTION
**Pull request by amishas157**
<details>
  <summary>PR Checklist</summary>

### PR Structure

- [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
- [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
      otherwise).
- [ ] This PR's title starts with the jira ticket associated with the PR.

### Thoroughness

- [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
- [ ] I've updated the README with the added features, breaking changes, new instructions on how to use the repository.

</details>

### What

This PR adds a new github workflow that:
- allows to select branch and environment
- Deploy to dev/prod (in this PR, we are only triggering dev deploy) . I will create a follow up PR for prod deploy steps

This also turns off dev-to-deploy step as it is not really used currently.

### Why

Improve the release workflow.

### Known limitations

None

### CODE REVIEW

File: .github/workflows/ci-cd-dev.yml [hunk 1]
```diff
@@ -66,59 +66,59 @@ jobs:
...
-          PROJECT_ID: test-hubble-319619
-          APPLICATION_CREDENTIALS: "${{ secrets.CREDS_TEST_HUBBLE }}"
-          COMPOSER_ENVIRONMENT: test-hubble-2
-          LOCATION: us-central1
-        with:
-          args: >
-            components install kubectl && gcloud composer environments run
-            $COMPOSER_ENVIRONMENT --location $LOCATION variables import
-            -- gcsfuse/actual_mount_path/variables.json
+  # deploy-to-dev:
```
Comment (chowbao) [inline .github/workflows/ci-cd-dev.yml:75]: Is this commented code copy and pasted to the new cicd? If so do you think it's worthwhile to keep it commented here or just delete it? We can also always look at git commit history for the file

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
+name: Deploy
+
+on:
+  workflow_dispatch:
+    inputs:
+      envName:
```
Comment (chowbao) [inline .github/workflows/deploy.yml:6]: Just wanted to confirm; does `envName` get converted to `inputs.env_name`?

File: .github/workflows/ci-cd-dev.yml (hunk 1 above)
Comment (amishas157) [reply .github/workflows/ci-cd-dev.yml:75]: yep it is part of new workflow. We can 🔪 it for once

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (amishas157) [reply .github/workflows/deploy.yml:6]: Uff no. That's my bad. Let me change it

File: .github/workflows/ci-cd-dev.yml (hunk 1 above)
Comment (amishas157) [reply .github/workflows/ci-cd-dev.yml]: Addressed in https://github.com/stellar/stellar-etl-airflow/pull/462/commits/bc4e5633c27c9f31a50815fdae3f35dee40c35b8

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (amishas157) [reply .github/workflows/deploy.yml:6]: Addressed in https://github.com/stellar/stellar-etl-airflow/pull/462/commits/bc4e5633c27c9f31a50815fdae3f35dee40c35b8

**Review by chowbao — APPROVED** [review]

File: .github/workflows/ci-cd-dev.yml [hunk 3]
```diff
@@ -45,99 +44,3 @@ jobs:
...
-          credentials_json: "${{ secrets.CREDS_TEST_HUBBLE }}"
-
-      - id: "get-credentials"
-        uses: "google-github-actions/get-gke-credentials@v2"
-        with:
-          cluster_name: "us-central1-test-hubble-2-5f1f2dbf-gke"
-          location: "us-central1"
-
-      - name: Pytest
-        run: pytest dags/
```
Comment (sydneynotthecity) [inline .github/workflows/ci-cd-dev.yml:73]: I don't know how useful it is to keep the `tests` job if we aren't actually running the Airflow unit tests. 
In the current workflow definition, the first time that the Airflow unit tests will execute is on deploy. Is that intended?

**Review by sydneynotthecity — COMMENTED** [review]
@amishas157 changes are looking good. I did have an overall design question. In your design doc, it proposed that deploys to either env (test, prod) should only come from the `master` branch. This PR, however, allows the developer to deploy to test off of any branch as needed (no review required). 

What were your thoughts on the dev workflow? 

- Dev commits feature branch
- when ready, selects branch name and deploys to test
- deploy ci executes

Is this correct?

**Comment by amishas157** [comment]
> changes are looking good. I did have an overall design question. In your design doc, it proposed that deploys to either env (test, prod) should only come from the master branch. This PR, however, allows the developer to deploy to test off of any branch as needed (no review required).
What were your thoughts on the dev workflow?
Dev commits feature branch
when ready, selects branch name and deploys to test
deploy ci executes
Is this correct?

Yes, that's still correct. I am currently using this PR to test my changes by deploying to dev. Unfortunately, github actions does not allow testing workflow before merging.

One the tests look good, my plan is to create followup PR to:
- Use if / else based on envName selected and deploy corresponding env
- Block prod deploy from non-master branch

File: .github/workflows/ci-cd-dev.yml (hunk 3 above)
Comment (amishas157) [reply .github/workflows/ci-cd-dev.yml:73]: So, my plan is to remove ci-cd.yml completely. However. the `tests` job is a required workflow to merge PR. Therefore, I kept it such that it passes.
Once the testing for `deploy.yml` is done, I will remove this file altogether

Comment (sydneynotthecity) [reply .github/workflows/ci-cd-dev.yml:73]: ahhhhh okie

**Comment by sydneynotthecity** [comment]
Sounds good, thanks for clarifying

**Pull request merged**